lib/scheduler/configmap.go

```diff
@@ -3,45 +3,75 @@ package scheduler
 import (
 	"encoding/json"
 	"fmt"
+	"sync"
 )
 
+// ConfigMap can contain mixed scheduler config types
+type ConfigMap map[string]Config
+
+// ConfigConstructor is a simple function that returns a concrete Config instance
+// with the specified name and all default values correctly initialized
+type ConfigConstructor func(name string, rawJSON []byte) (Config, error)
+
+var (
+	configTypesMutex   sync.RWMutex
+	configConstructors = make(map[string]ConfigConstructor)
+)
+
+// RegisterConfigType adds the supplied ConfigConstructor as the constructor for its
+// type in the configConstructors map, in a thread-safe manner
+func RegisterConfigType(configType string, constructor ConfigConstructor) {
```

Comment:
Maybe just `Register` like in the `sql` package I doubt there is need to be this specific - we are not going to register anything else and `scheduler.Register` is in my opinion the correcter name than `scheduler.RegisterConfigType`

Reply:
Hm maybe I should have renamed the package to `lib.schedulerconfig` or something like that, since it's currently only `scheduler` and the objects inside of it have a `Config` suffix. But I like the current approach a bit better, since I can also put other scheduler related helper functions here, and I can differentiate between the actual scheduler implementations and their configs. I'd like to leave this as it is for now, but I may end up renaming it later, with the following PRs...